pkg/provider/azure_standard.go

```diff
@@ -1004,11 +1005,14 @@ func (as *availabilitySet) EnsureBackendPoolDeleted(service *v1.Service, backend
 	for i := range ipConfigurationIDs {
 		ipConfigurationID := ipConfigurationIDs[i]
 		nodeName, _, err := as.GetNodeNameByIPConfigurationID(ipConfigurationID)
-		if err != nil {
+		if err != nil && !errors.Is(err, cloudprovider.InstanceNotFound) {
 			klog.Errorf("Failed to GetNodeNameByIPConfigurationID(%s): %v", ipConfigurationID, err)
 			allErrs = append(allErrs, err)
 			continue
 		}
+		if nodeName == "" {
+			continue
```

Comment:
also remove this block?

Reply:
we should check and continue for next ipConfigurationID because without a nodeName, the following queries for VM and NIC won't succeed and hence couldn't do any NIC put operations later.